src/components/Fuel/FuelForm.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useState } from 'react';
+import { Button, Label, Select, TextInput } from 'flowbite-react';
+import {
+  FuelSourceUnit,
+  FuelSourceType,
+  UnregisteredFuelRequest,
+} from '../../schema/fuel.schema';
+
+interface FuelFormProps {
+  handleSubmit: (fuelData: UnregisteredFuelRequest) => void;
+}
+
+export default function FuelForm({ handleSubmit }: FuelFormProps) {
+  /* Initializing state */
+  const [fuelData, setfuelData] = useState({
+    fuel_source_type: 'ng' as FuelSourceType,
+    fuel_source_unit: 'btu' as FuelSourceUnit,
+    fuel_source_value: 0 as number,
+  });
+
+  function submitForm(e: React.FormEvent) {
+    e.preventDefault();
+    console.log(fuelData);
+    handleSubmit(fuelData);
+  }
+
+  /* Function that updates the state */
+  function handleChange(event: React.ChangeEvent<HTMLInputElement>) {
```

Comment:
Change `React.ChangeEvent<HTMLInputElement>` to `React.ChangeEvent<HTMLInputElement | HTMLSelectElement>`

Reply:
I tried it at first, but since I have both input and select in my state I just traded one type correction for another.

Reply:
Yeah, you'll need to enact the other changes I suggested for it to all work.